packages/lesswrong/server/callbacks/votingCallbacks.ts

```diff
@@ -130,3 +138,121 @@ postPublishedCallback.add(async (publishedPost: DbPost) => {
+// on it making top 50 in the review, and create a comment linking to the market.
+
+const manifoldAPIKeySetting = new DatabaseServerSetting<string | null>('manifold.reviewBotKey', null)
+const manifoldAPIKey = manifoldAPIKeySetting.get()
+
+const reviewUserBotSetting = new DatabaseServerSetting<string | null>('reviewBotId', null)
+const reviewUserBot = reviewUserBotSetting.get()
+
+async function addTagToPost(postId: string, tagSlug: string, botUser: DbUser, context: ResolverContext) {
+  const tag = await Tags.findOne({slug: tagSlug})
+  if (!tag) throw new Error(`Tag with slug "${tagSlug}" not found`)
+  await addOrUpvoteTag({tagId: tag._id, postId: postId, currentUser: botUser, context});
+}
+
+// AFAIU the flow, this has a race condition. If a post is voted on twice in quick succession, it will create two markets.
+// This is probably fine, but it's worth noting. We can deal with it if it comes up.
+voteCallbacks.castVoteAsync.add(async ({newDocument, vote}: VoteDocTuple, collection, user, context) => {
+
+  // Forum gate
+  if (!isLWorAF) return;
+
+  if (!manifoldAPIKey) throw new Error("Manifold API key not found");
+  if (!reviewUserBot) throw new Error("Review bot user not found");
+
+  if (collection.collectionName !== "Posts") return;
+  if (vote.power <= 0 || vote.cancelled) return; // In principle it would be fine to make a market here, but it should never be first created here
+  if (newDocument.baseScore < MINIMUM_KARMA_REVIEW_MARKET_CREATION) return;
+  const post = await Posts.findOne({_id: newDocument._id})
+  if (!post) return;
+  if (post.postedAt.getFullYear() < (new Date()).getFullYear() - 1) return; // only make markets for posts that haven't had a chance to be reviewed
+  if (post.manifoldReviewMarketId) return;
+
+  const botUser = await context.Users.findOne({_id: reviewUserBot})
+  const annualReviewLink = 'https://www.lesswrong.com/tag/lesswrong-review'
+  const postLink = 'https://www.lesswrong.com/posts/' + post._id + '/' + post.slug
+
+  const year = post.postedAt.getFullYear()
+  const initialProb = 14
+  const question = `Will "${post.title.length < 50 ? post.title : (post.title.slice(0,45)+"...")}" make the top fifty posts in LessWrong's ${year} Annual Review?`
+  const descriptionMarkdown = `As part of LessWrong's [Annual Review](${annualReviewLink}), the community nominates, writes reviews, and votes on the most valuable posts. Posts are reviewable once they have been up for at least 12 months, and the ${year} Review resolves in February ${year+2}.\n\n\nThis market will resolve to 100% if the post [${post.title}](${postLink}) is one of the top fifty posts of the ${year} Review, and 0% otherwise. The market was initialized to ${initialProb}%.` // post.title
+  const closeTime = new Date(year + 2, 1, 1) // i.e. february 1st of the next next year (so if year is 2022, feb 1 of 2024)
+  const visibility = isProduction ? "public" : "unlisted"
+
+  if (!botUser) throw new Error("Bot user not found")
```

Comment:
We should perhaps move this up to immediately after attempting to find the bot user?